File: neel/D_Antiamuny_Wants_to_Learn_Swap.py
t = int(input())

for _ in range(t):
    n, q = map(int, input().split()) 
    a = list(map(int, input().split()))

    for _ in range(q):
        l, r = map(int, input().split())
        l -= 1
        r -= 1
        nee = a[l:r+1]
        jee = sorted(nee)

        i = 0
        while i < len(nee)-1:
            if nee[i] < nee[i+1]:
                i += 1
            else:
                nee[i], nee[i+1] = nee[i+1], nee[i]
                i += 2

        if nee == jee: 
            print("YES")
        else:
            print("NO")

File: neel/D_Antiamuny_Wants_to_Learn_Swap.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int t;
    cin >> t;

    while (t--) {
        int n, q;
        cin >> n >> q;

        vector<int> a(n);
        for (int i = 0; i < n; i++) {
            cin >> a[i];
        }

        while (q--) {
            int l, r;
            cin >> l >> r;
            l--, r--;

            vector<int> nee(a.begin() + l, a.begin() + r + 1);
            vector<int> jee = nee;
            sort(jee.begin(), jee.end());

            int i = 0;
            while (i < (int)nee.size() - 1) {
                if (nee[i] < nee[i + 1]) {
                    i++;
                } else {
                    swap(nee[i], nee[i + 1]);
                    i += 2;
                }
            }

            if (nee == jee) {
                cout << "YES\n";
            } else {
                cout << "NO\n";
            }
        }
    }

    return 0;
}
